### PR TITLE
Compliance with PEP440, PEP517, and PEP518

### PR DESCRIPTION
**Pull request by robertodr**
### TLDR

This PR allows to:
- build and install VAMPyR through `pip`.
- allow Sphinx to correctly extract documentation in the compiled Python module. The docs should be written in the bindings themselves, see the binding for `mrcpp_version` for an example.
- fix the versioning number to follow [PEP440](https://www.python.org/dev/peps/pep-0440/).

----

Recent versions of pip can build compiled extensions, such that, locally, one can do after cloning:
```
$ python -m pip install .
```
the dependencies for installing can be specified in the `pyproject.toml` and `setup.py` (the canonical way of packaging Python modules) knows how to interact with that. For the details see [PEP517](https://www.python.org/dev/peps/pep-0517/) and [PEP518](https://www.python.org/dev/peps/pep-0518/). For extensions built with CMake, one needs some more "magic" available through [`scikit-build`](https://scikit-build.readthedocs.io/en/latest/).

To achieve this, I:

- Had to move the sources from `vampyr` to `src/vampyr`.
- Had to rename the version in `VERSION` to comply with PEP440. `pip` will refuse to build otherwise.
- Removed Autocmake and the `setup` script, because:
   * The modules we use from Autocmake are stable and there's no reason to keep the dependency.
   * There haven't been much updates to Autocmake and it's a bit contrived to use it to generate the root `CMakeLists.txt`.
   * It's confusing to have `setup` and `setup.py`. Furthermore, `setup` doesn't do much more than call `cmake` and that can be done easily by hand:
      ```
      $ cmake -S. -Bbuild
      ``` 
* Bumped CMake dependency and refactored some modules here and there.
* Create `__version__` from the C++ binding, rather than in `__init__.py` such that we avoid having to configure the latter with CMake (it's troubling for `setup.py` otherwise)

Miscellaneous changes:
* Reformat Python sources with black and isort.
* Added documentation for the root module to its C++ binding.
* Copy-pasted the changes in the `documentation` branch.

### CODE REVIEW

**Comment by robertodr** [comment]
To pass arguments to CMake and/or ninja/make you have to install using
`setup.py` directly:
https://scikit-build.readthedocs.io/en/latest/usage.html#command-line-options

For example:

```
$ python setup.py install -- -DMRCPP_DIR=/some/where -- -j4
```

Note that the current setup uses Ninja, 4 cores to build and 2 cores to
link, so the last bit `-- -j4` is unnecessary. I have not tested the above
on vampyr, but it did work on another project.

**Comment by robertodr** [comment]
Now we only need to write the docstrings in the binding source code 😸
